core-navigation/src/main/java/com/google/samples/apps/nowinandroid/core/navigation/NiaNavigationDestination.kt

```diff
@@ -36,3 +36,11 @@ interface NiaNavigationDestination {
      */
     val destination: String
 }
+
+interface NiaNavigationWithArguments {
```

Comment:
Having another interface just for this seems like an overkill. You could just add:

`val routeWithArgs = "$route/{$topicIdArg}"`
` val routeWithArgs = "$route/{$authorIdArg}"`

To TopicDestination and AuthorDestination accordingly.

Reply:
I think the interface is different. There needs to be a separation between destination and route with arguments. By making the interface usable in multiple navigations, it eliminates repetitive write variables.